packages/core/src/transport/transport.ts

```diff
@@ -1,11 +1,45 @@
 import { display } from '../tools/display'
 import { Context } from '../tools/context'
 import { addEventListener, DOM_EVENT, jsonStringify, noop, objectValues } from '../tools/utils'
-import { monitor, addErrorToMonitoringBatch } from '../domain/internalMonitoring'
+import { monitor, addErrorToMonitoringBatch, addMonitoringMessage } from '../domain/internalMonitoring'
 
 // https://en.wikipedia.org/wiki/UTF-8
 const HAS_MULTI_BYTES_CHARACTERS = /[^\u0000-\u007F]/
 
+/* eslint-disable camelcase */
+type ErrorCtx =
+  | {
+      err_msg?: string
+      err_stack?: string
+    }
+  | undefined
+
+type TransportErrorCtx = {
+  beacon: Record<string, string | number | boolean>
+  event?: Record<string, string | number | boolean>
+  req?: Record<string, string | number | boolean>
+  error_ctx?: ErrorCtx
+  on_line: boolean
+}
+/* eslint-enable camelcase */
+
+const getTransportErrorEventDetails = (evt: ProgressEvent) => {
```

Comment:
We tend to put the most important functions (ex: the exported functions) at the top of modules. Could you move this one a bit below?